Upgrade demoshop to beta4 version

grossmannmartin opened the pull request:
| Q             | A
| ------------- | ---
|Description, reason for the PR| Keep demoshop upgraded with newest version of Framework
|New feature| No <!-- Do not forget to update CHANGELOG.md and possibly docs/ -->
|BC breaks| No <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| -
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

grossmannmartin committed:
updated Docker configuration

grossmannmartin committed:
renamed Database tests to Functional tests

- Class DatabaseTestCase was renamed to TransactionFunctionalTestCase (see shopsys/shopsys#541)

grossmannmartin committed:
phpUnit: Remove unsupported syntaxCheckAttribute

- see shopsys/shopsys#592

grossmannmartin committed:
unnecessary files in docker are now ignored

grossmannmartin committed:
content-test directory is used instead of content during the test

- see shopsys/shopsys#547

grossmannmartin committed:
updated coding standards and applied

- added missing @param and @return annotations through the whole codebase
- see shopsys/shopsys#573

grossmannmartin committed:
dependencies on shopsys packages updated to beta3 version

grossmannmartin committed:
domains URLs are auto-configured during "composer install"

- see shopsys/shopsys#540

grossmannmartin committed:
new form types are now registered

- new OrderItemsType for displaying table for order items management
- new DisplayOnlyCustomerType for displaying information about Customer
- see shopsys/shopsys#576

grossmannmartin committed:
phing targets: create-domains-data is now dependent on create-domains-db-functions

grossmannmartin committed:
phing: improve db-fixtures targets (dependency order, remove obsolete, rename)

- remove db-fixtures-demo-multidomain and test-db-fixtures-demo-multidomain and their usages
- switch dependency order of db-demo to ...,create-domains-data,db-fixtures-demo-singledomain,...
- switch dependency order of test-db-demo to ...,test-create-domains-data,test-db-fixtures-demo-singledomain,...
- change db-fixtures-demo-singledomain and test-db-fixtures-demo-singledomain command to <arg value="doctrine:fixtures:load" /> as command shopsys:fixtures:load doesn't exist anymore and remove --fixtures argument
- rename db-fixtures-demo-singledomain to db-fixtures-demo
- rename test-db-fixtures-demo-singledomain to test-db-fixtures-demo

grossmannmartin committed:
OrderFormType in Admin is now properly extended

- there is no need to extend twig template anymore
- see shopsys/shopsys#576

grossmannmartin committed:
fixed tests related with price calculation removal

grossmannmartin committed:
changed Shopsys\Environment class for consistent env setting during composer install

- see shopsys/project-base@eedcf2ea9eaaef6c4f53a83fedbbd3c34428af83

grossmannmartin committed:
dependencies on shopsys packages updated to beta4 version

vitek-rostislav commented on docker/conf/docker-compose-mac.yml.dist:
```diff
@@ -88,36 +75,27 @@ services:
 
     microservice-product-search:
-        image: shopsys/microservice-product-search:v7.0.0-beta2
+        image: shopsys/microservice-product-search:latest
```
should not this be `v7.0.0-beta4`? I believe our upgrading notes are not mentioning this at all, and they link to master versions of the files, however, when upgrading, I think we should use the tagged versions (similarly to changes in `composer.json`)

grossmannmartin replied:
Yes, you're right. Fixed!

vitek-rostislav commented on tests/ShopBundle/Functional/Component/Doctrine/GroupedScalarHydratorTest.php:
```diff
@@ -1,14 +1,14 @@
+use Tests\ShopBundle\Test\TransactionFunctionalTestCase;
 
-class GroupedScalarHydratorTest extends DatabaseTestCase
+class GroupedScalarHydratorTest extends TransactionFunctionalTestCase
```
it would be nice to mention in the commit message that `DatabaseTestCase` has been renamed to `TransactionFunctionalTestCase`

vitek-rostislav commented on docker/conf/docker-sync.yml.dist:
```diff
@@ -3,4 +3,20 @@ syncs:
         sync_userid: 501
         src: './'
-        sync_excludes: ['docker/']
+        sync_excludes: sync_excludes: [
```
this is weird, you have the `sync_excludes` twice here, probably a typo?

grossmannmartin replied:
Yes, it's a typo. Fixed!

vitek-rostislav commented on app/config/paths.yml:
```diff
@@ -1,33 +1,34 @@
     shopsys.javascript_sources_dir: '%shopsys.resources_dir%/scripts'
-    shopsys.javascript_url_prefix: /assets/scripts/
-    shopsys.log_stream: /tmp/log-pipe
+    shopsys.javascript_url_prefix: '/assets/scripts/'
```
Was that necessary to do? Moreover, strictly speaking, these changes should be done in a separate commit.

grossmannmartin replied:
@vitek-rostislav Not exactly necessary. When I've changed `/content/` to a parameter, I had to add quotes to these values. 
And I end up unifying all the values. You're right, that not affected lines could be in a separate commit for clarity.

grossmannmartin replied:
Is it ok to leave it like this, or should I split it?

vitek-rostislav replied:
Yes, it is ok to leave it as it is now...but next time... :slightly_smiling_face:

vitek-rostislav commented on docker/php-fpm/Dockerfile:
```diff
@@ -58,21 +60,57 @@ ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
+
+RUN composer install --optimize-autoloader --no-interaction --no-progress
+
+RUN php phing composer-dev npm dirs-create test-dirs-create assets standards tests-static tests-acceptance-build
```
The `test-dirs-create` should be added in https://github.com/shopsys/demoshop/pull/27/commits/c31f45ce777bd8a580538187111a5896b6a13647. The phing target does not exist at this moment, However, I see that it would be very difficult to update the Dockerfile gradually - you do not need to fix it here, we would just need to think about it when improving the upgrading process and instructions.

grossmannmartin replied:
I agree. Upgrade instructions can prevent this kind of mistakes.

vitek-rostislav commented on composer.json:
```diff
@@ -96,7 +96,7 @@
         "phpstan/phpstan": "^0.7",
         "phpunit/phpunit": "^7.0",
-        "shopsys/coding-standards": "v7.0.0-beta2",
+        "shopsys/coding-standards": "v7.0.0-beta3",
```
It seems to me like a good idea to solve the standards first :+1:

vitek-rostislav replied:
I would just slightly edit the 2nd line of the commit message:

"- added missing @param and @return annotations through the whole codebase"

vitek-rostislav commented on composer.json:
```diff
@@ -68,15 +68,15 @@
-        "shopsys/form-types-bundle": "v7.0.0-beta2",
-        "shopsys/framework": "v7.0.0-beta2",
-        "shopsys/plugin-interface": "v7.0.0-beta2",
+        "shopsys/migrations": "v7.0.0-beta3",
```
"deps" - please, try not to use abbreviations in the commit messages :slightly_smiling_face:

vitek-rostislav replied:
I would say "dependencies on shopsys packages updated to beta3 version" to be more descriptive

vitek-rostislav commented on build.xml:
```diff
@@ -271,17 +271,6 @@
         </exec>
     </target>
 
-    <target name="domains-urls-check" description="Checks the domain urls configuration.">
```
I am thinking that it would be probably nice to link the original pull requests on the 2nd line of your commit message :slightly_smiling_face: 

"- see https://github.com/shopsys/shopsys/pull/540"

vitek-rostislav commented on app/config/packages/twig.yml:
```diff
@@ -7,7 +7,9 @@ twig:
         - '@ShopsysFramework/Admin/Form/fileuploadFields.html.twig'
         - '@ShopsysFramework/Admin/Form/imageuploadFields.html.twig'
         - '@ShopsysFramework/Admin/Form/pricetableFields.html.twig'
+        - '@ShopsysFramework/Admin/Form/orderItems.html.twig'
```
again, link to the original PR would be probably helpful when someone was trying to understand this change

vitek-rostislav commented on build-dev.xml:
```diff
@@ -16,7 +16,7 @@
     <target name="standards" depends="phplint,ecs,phpstan,twig-lint,yaml-lint,eslint-check" description="Checks coding standards."/>
```
you need to be more descriptive in commit message "phing target: Make couple of changes" - I do not know what happened and why :wink:

vitek-rostislav commented on tests/ShopBundle/Functional/Model/Cart/Watcher/CartWatcherServiceTest.php:
```diff
@@ -4,58 +4,51 @@
+        /** @var \Shopsys\ShopBundle\Model\Product\Product $product */
+        $product = $this->getReference(ProductDataFixture::PRODUCT_PREFIX . '1');
 
+        /* @var \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPriceCalculationForUser $productPriceCalculationForUser */
```
should be "/** @var..." :slightly_smiling_face:

vitek-rostislav replied:
however, this is inconsistent throughout multiple classes here, so I would not bother :wink:

grossmannmartin replied:
It would be nice to unify annotations. I've opened an issue to address this problem (#29)

vitek-rostislav commented on tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php:
```diff
@@ -14,20 +13,17 @@
 {
     public function testOnKernelResponseNoAction()
     {
+        /** @var \Shopsys\FrameworkBundle\Component\Setting\Setting $setting */
```
I wonder, how come that in some cases, you fixed these annotations? :laughing:

grossmannmartin replied:
Yes, when I rewrote Tests, I tried to unify annotations, so it's easier to read a code. 
This should be unified (#29)

vitek-rostislav reviewed (changes requested):
Hi @grossmannmartin, good job :+1: I have added a few notes so please check them out :slightly_smiling_face:

As I mentioned in some of the notes, I think it would be nice to link the original pull requests in the commit messages where it makes sense. I could help with understanding the changes in the project.

vitek-rostislav reviewed (approved):
Hi, the fixes are ok with me :+1: Let me know after the commit messages are improved :slightly_smiling_face:

vitek-rostislav reviewed (approved):
Hi @grossmannmartin, all the changes are now ok with me :+1: After the tests are done, you can move the task to business validation :tada:

boris-brtan commented on build-dev.xml:
```diff
@@ -580,38 +580,11 @@
-                    <arg value="--env=test" />
-                    <arg value="shopsys:fixtures:load" />
-                    <arg value="--fixtures=${path.framework}/src/DataFixtures/DemoMultidomain" />
-                    <arg value="--fixtures=${path.src}/Shopsys/ShopBundle/DataFixtures/DemoMultidomain" />
```
hi, what do we want to do with upgrade of datafixtures ?

vitek-rostislav replied:
oh yes :man_facepalming: Boris, thanks for noticing, we need to fix that :+1:

grossmannmartin replied:
Data fixtures resolved in f86cf2dc2a028dbf2f5fa0e81ccec0954958c499

boris-brtan replied:
OK, i see still 2 folders in Datafixtures but maybe it is ok for different project with different implementation as project-base

grossmannmartin committed:
block loading of original data fixtures, if overridden

- overriden data fixtures can be loaded before original, resulting in unwanted data and
block loading of original data fixtures fix this problem
- see shopsys/shopsys#568

boris-brtan commented on docker/conf/docker-compose-mac.yml.dist:
```diff
@@ -88,36 +75,27 @@ services:
 
     microservice-product-search-export:
-        image: shopsys/microservice-product-search-export:v7.0.0-beta2
+        image: shopsys/microservice-product-search-export:beta4
```
please fix

grossmannmartin replied:
Thanks for noticing. Fixed!

grossmannmartin committed:
webserver container now starts after php-fpm is started

- see shopsys/shopsys#679

grossmannmartin committed:
fixed wrong links in mail template data fixtures

grossmannmartin committed:
updated changelog with beta4 upgrade informations
